framework/include/mesh/MooseMesh.h

```diff
@@ -818,6 +818,12 @@ class MooseMesh : public MooseObject, public Restartable
    */
   virtual std::unique_ptr<PointLocatorBase> getPointLocator() const;
 
+  /**
+   * Returns the name of the mesh file read to produce this mesh if any or an empty string
+   * otherwise.
+   */
+  virtual std::string getFileName() const { return ""; }
```

Comment:
This seems like a reasonable new API to add, but I know we have several other mesh formats that read files spread all over (there maybe another in the framework, there are several in the modules, etc). You should probably take a look at make them all override this new interface.

Reply:
I'll take a look.  Not sure what to do about the meshes that read multiple files though.

Reply:
This is done.